homeassistant/components/life360/device_tracker.py

```diff
@@ -0,0 +1,617 @@
+})
+
+PLATFORM_SCHEMA = vol.All(LIFE360_SCHEMA, _thresholds)
+
+
+def _include_name(filter_dict, name):
+    if not name:
+        return False
+    if not filter_dict:
+        return True
+    name = name.lower()
+    if filter_dict['include']:
+        return name in filter_dict['list']
+    return name not in filter_dict['list']
+
+
+def _exc_msg(exc):
+    return '{}: {}'.format(exc.__class__.__name__, str(exc))
+
+
+def _setup_zone_updating(hass, config, api):
+    from life360 import Life360Error
+
+    add_zones = config.get(CONF_ADD_ZONES)
+    if not add_zones:
+        return
+
+    Place = namedtuple(
+        'Place', [ATTR_NAME, ATTR_LATITUDE, ATTR_LONGITUDE, ATTR_RADIUS])
+
+    def get_places(api, circles_filter, places_filter):
+        errs = 0
+        while True:
+            places = set()
+            try:
+                for circle in api.get_circles():
+                    circle_id = circle['id']
+                    if not (circle_id
+                            and _include_name(circles_filter, circle['name'])):
+                        continue
+                    for place in api.get_circle_places(circle_id):
+                        place_name = place['name']
+                        if not _include_name(places_filter, place_name):
+                            continue
+                        places.add(Place(place_name,
+                                         float(place['latitude']),
+                                         float(place['longitude']),
+                                         float(place['radius'])))
+            except (Life360Error, KeyError, TypeError, ValueError) as exc:
+                errs += 1
+                if errs >= 3:
+                    _LOGGER.error('get_places: %s', _exc_msg(exc))
+                    return None
+            else:
+                return places
+
+    def log_places(msg, places):
+        plural = 's' if len(places) > 1 else ''
+        _LOGGER.debug(
+            '%s zone%s from Place%s: %s',
+            msg, plural, plural,
+            '; '.join(['{}: {}, {}, {}'.format(*place) for place
+                       in sorted(places, key=lambda x: x.name.lower())]))
+
+    def zones_from_places(api, circles_filter, places_filter, home_place_name,
+                          zones):
+        _LOGGER.debug('Checking Places')
+        places = get_places(api, circles_filter, places_filter)
+        if places is None:
+            return
+
+        # See if there is a Life360 Place whose name matches CONF_HOME_PLACE.
+        # If there is, remove it from set and handle it specially.
+        home_place = None
+        for place in places:
+            if place.name.lower() == home_place_name:
+                home_place = place
+                places.discard(place)
+                break
+
+        # If a "Home Place" was found and it is different from the current
+        # zone.home, then update zone.home with it.
+        if home_place:
```

Comment:
Let's drop this and open an arch issue to discuss updating home zone.

Reply:
Honestly, if I drop this then I'd rather drop the entire feature of creating/updating HA zones based on Life360 Places. This is the main reason for the feature existing in the first place.

I can guarantee, based on experience with the existing custom component, without this feature there's going to be lots of confusion when HA and Life360 don't agree on the user being home. It's happened many times already, and it will just happen more if/when this is introduced as a standard integration.

Actually, I'd rather _not_ make it a standard integration without this feature and just live with maintaining the custom component.

It's up to you. If you want to discuss this in an arch issue, I'll just wait for that outcome. I personally see nothing wrong with an integration updating HA's data. That's basically what HA does -- take data from other systems, and in the process, make it more useful. I don't see why this should be any different.

Reply:
Okay, in that case let's keep it a custom integration and close this PR.

Reply:
@balloob ok, I'll go back to the drawing board and try to come up with something simpler.